presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java

```diff
@@ -181,6 +197,19 @@ public TpchTableHandle getTableHandle(ConnectorSession session, SchemaTableName
                             .collect(toSet())));
             unenforcedConstraint = filterOutColumnFromPredicate(constraint.getSummary(), OrderColumn.ORDER_STATUS);
         }
+        else if (tableHandle.getTableName().equals(TpchTable.PART.getTableName())) {
+            predicate = toTupleDomain(ImmutableMap.of(
+                    toColumnHandle(PartColumn.CONTAINER),
+                    PART_CONTAINER_NULLABLE_VALUES.stream()
+                            .filter(convertToPredicate(constraint.getSummary(), PartColumn.CONTAINER))
+                            .collect(toSet()),
+                    toColumnHandle(PartColumn.TYPE),
+                    PART_TYPE_NULLABLE_VALUES.stream()
+                            .filter(convertToPredicate(constraint.getSummary(), PartColumn.TYPE))
+                            .collect(toSet())));
```

Comment:
These hardly readable 9 lines constructing tuple domain contain just two pieces of information:
- `PartColumn.TYPE` has `PART_TYPE_NULLABLE_VALUES` values
- `PartColumn.CONTAINER` has `PART_CONTAINER_NULLABLE_VALUES`.


What about:
```
ImmutableMap.builder()
 .put(toColumnHandle(PartColumn.CONTAINER), filterValuesForColumn(
   PART_CONTAINER_NULLABLE_VALUES,
   constraint.getSummary(),
   PartColumn.CONTAINER))
 .put(toColumnHandle(PartColumn.TYPE), filterValuesForColumn(
   PART_TYPE_NULLABLE_VALUES,
   constraint.getSummary(),
   PartColumn.TYPE))
.build()
```

(or somehow further reduce boilerplate)